samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -134,10 +134,11 @@ def __init__(
         self._debugger_path = debugger_path
         self._container_env_vars_file = container_env_vars_file
 
-        self.parameter_overrides = parameter_overrides or {}
+        self._parameter_overrides = parameter_overrides
         # Override certain CloudFormation pseudo-parameters based on values provided by customer
+        self._pseudo_parameter_overrides: Optional[Dict] = None
```

Comment:
why do we need this additional property? why not just adding the AWS::Region to self._parameter_overrides

Reply:
@mndeveci asked the same question above: https://github.com/aws/aws-sam-cli/pull/2631#discussion_r582447560